state-chain/pallets/cf-threshold-signature/src/migrations/migrate_signature_to_include_signer.rs

```diff
@@ -25,9 +25,8 @@ mod old {
 
 impl<T: crate::Config<I>, I: 'static> OnRuntimeUpgrade for Migration<T, I> {
 	fn on_runtime_upgrade() -> frame_support::weights::Weight {
-		let current_aggkey =
-			Keys::<T, I>::get(CurrentKeyEpoch::<T, I>::get().unwrap_or(Default::default()))
-				.unwrap_or(Default::default());
+		let current_aggkey = Keys::<T, I>::get(CurrentKeyEpoch::<T, I>::get().unwrap_or_default())
```

Comment:
Why are we defaulting here? When we do expect Keys for the current epoch to be empty?

Reply:
So the idea is that there is no situation where the Keys would be empty on a running network except for the situation where a chain is not initialised in which case we wont have anything to migrate anyways so the default wont matter. Practically speaking, this situation can only occur with solana code being active on a chain (and solana not being initialized yet). To safeguard against that, the migration for the solana broadcast pallet is not even run. See the runtime lib file.

Reply:
I can maybe see the value in using default for the actual runtime upgrade just for very edge cases, we don't want it to panic for the non-solana chains.

However, for the pre and post checks we should not unwrap_or_default. There's current a hole in the test.

On runtime upgrade runs, we don't have a key, so we insert the default key.

Now on the post check we try to get a key and we get the default because it doesn't exist... now we're just comparing two default keys and the test *Passes*. This test should fail if we expect a key.

Reply:
> I can maybe see the value in using default for the actual runtime upgrade just for very edge cases, we don't want it to panic for the non-solana chains.

As said before, for non-solana chains, there is no edge case on a running network since there is no scenario that Keys would be empty so not sure what you mean here.

> However, for the pre and post checks we should not unwrap_or_default. There's currently a hole in the test.
> 
> On runtime upgrade runs, we don't have a key, so we insert the default key.
> 
> Now on the post check we try to get a key and we get the default because it doesn't exist... now we're just comparing two default keys and the test _Passes_. This test should fail if we expect a key.

I don't agree with this. The case you are describing was already considered but as said before, is hypothetical. The tests wont run for the case you describe. The test for that case is different (see ```NoSolUpgrade```). Its no different than unwrapping a value giving a reason of why that is safe to do so. I would just ```unwrap``` here and not ```unwrap_or_default``` (which I did originally) but the ci upgrade test was panicing on the unwrap which, if anything, shows there is an issue with the ci upgrade test not the test for the migration.

Reply:
discussed irl. The ```unwrap_or_default()``` should be changed back to ```unwrap()``` since the default case wont happen as mentioned above. The ci upgrade test should pass with the ```NoSolUpgrade``` which wasnt there when the test was failing.